### PR TITLE
Add transfer_to to user model

### PR DESCRIPTION
**Pull request by benglish-sumo**
Add a transfer_to field to user resources to allow inheriting of content after user deletion

### CODE REVIEW

File: sumologic/resource_sumologic_user.go [hunk 1]
```diff
@@ -56,6 +56,11 @@ func resourceSumologicUser() *schema.Resource {
 				Optional: true,
 				ForceNew: false,
 			},
+			"transfer_to": {
```
Comment (seansain) [inline sumologic/resource_sumologic_user.go]: What happens if I do not specify someone to transfer to? Didn't we want the ability to send this to the account owner if no one is specified?

**Comment by seansain** [comment]
If you have not run the `tfproviderlint` make sure to do so. You can also rebase against master and the PR builder will automatically do it.

**Comment by benglish-sumo** [comment]
I wouldn't call that a bug, that's how terraform works. Any local change not applied has no effect on the state of the system.

Also, because of the way terraform initializes non required fields, it's impossible to differentiate between deliberately empty and empty by default. The only way I could see to do that is add a special value like "me" that the delete call translates into the tf user's userId and default to that.

We could also just make transfer_to required so any empty transfer is deliberate.

**Comment by seansain** [comment]
Ya, that is true, but I was saying that it is definitely something that will be confusing for the user, so it is something we should add to the documentation.
But I do like the idea of just making `transfer_to` required. Can you do that and add it to the docs: https://github.com/SumoLogic/sumologic-terraform-provider/blob/master/website/docs/r/user.html.markdown

**Comment by benglish-sumo** [comment]
This will require the addition of the transfer_to field to all user resources

**Pull request merged**